go/vt/vttablet/tabletserver/tx_pool.go

```diff
@@ -232,14 +232,19 @@ func (axp *TxPool) Begin(ctx context.Context, options *querypb.ExecuteOptions) (
 		return 0, err
 	}
 
-	if query, ok := txIsolations[options.GetTransactionIsolation()]; ok {
-		if _, err := conn.Exec(ctx, query, 1, false); err != nil {
+	if options.GetTransactionIsolation() == querypb.ExecuteOptions_CONSISTENT_SNAPSHOT_READ_ONLY {
```

Comment:
Will be better to change `txIsolations` to return a list of queries, including the `begin`. And we can issue the default `begin` only if `!ok`.